src/pbkdf2.rs

```diff
@@ -1,10 +1,94 @@
 
 use bitcoin_hashes::{hmac, sha512, Hash, HashEngine};
 
+const SALT_PREFIX: &'static str = "mnemonic";
+
+/// Calculate the binary size of the mnemonic.
+fn mnemonic_byte_len(mnemonic: &[&'static str]) -> usize {
+	let mut len = 0;
+	for i in 0..mnemonic.len() {
+		let word = &mnemonic[i];
+		if word.is_empty() {
```

Comment:
I wonder if it wouldn't be better to calculate the length in `Mnemonic` before calling the pbkdf function so that a slice with only non-empty words can be passed.

Reply:
Good point.